### PR TITLE
VTE submit data testscript + minimal fixtures

### PR DESCRIPTION
**Pull request by mgramigna**
``` bash
brake crucible:execute[<base stu3 url>,stu3,vte-submit-data]
```

Simple $submit-data testscript for the VTE measure. @okeefm gave me a minimal MeasureReport, Patient, and Encounter that should fall into the IPOP for this measure. Be sure to look at the parameters as well as the testscript.

### CODE REVIEW

**Review by okeefm — APPROVED** [review]
Looks good, considering we still don't have a proper evaluation of `VTE-1` on Kanvix. Tested on my local box and it passes.

**Review by tstrass — COMMENTED** [review]
Should we add in the assertion on evaluate-measure even though cqf-ruler doesn't support it yet so that this is good to go when it does, or did we decide we just want them passing for now?

Also, should this have teardown?

**Comment by mgramigna** [comment]
@tstrass re: teardown. I'm conflicted. This is what I posted in our slack related to the other TestScript update PR but it applies here too I think:

"Where do we stand on our teardown sections of TestScripts? The purpose of these is to delete the resource we created during setup before running our tests. There are no real problems when we create these resources and have the server autogenerate the ID, as we can just store the ID in the TestScript and then delete it afterward. The problem is when we have a resource (like a Measure) that references another resource (like a Library) explicitly by ID in that Measure’s JSON. We need to ensure that a Library with that specific ID is on the server so we can successfully POST the Measure. To do this, we need to use a PUT instead so we can control the ID of the LIbrary. This means that we are creating the Library if it doesn’t exist on the server, and updating it if it does. What should we do in the teardown section in this case? Ideally we would delete it if it was created, and not delete it if it was updated, but there is no nice way to do this in the TestScript itself. If we’re only deleting some of the resources during teardown but not all, is it even worth it to delete anything?

TL;DR: Is it worth deleting stuff with the teardown section of TestScripts if there are some cases where we don’t want to delete resources that already existed on the server?"

**Review by tstrass — CHANGES_REQUESTED** [review]
Per our discussion on slack, teardown should be added and and resource IDs should be prefixed with a "namespace" to ensure we don't tamper with any data we don't own.

**Comment by mgramigna** [comment]
I implement the "MitreTestScript-" prefixes to our test fixtures and added a proper teardown section. I think we need to have short discussion about the measure evaluation stuff as well, so I'll follow up on that tomorrow.

**Review by tstrass — CHANGES_REQUESTED** [review]
Running this again, it passed the first time, but subsequent attempts fail with a 400 error on the `$submit-data` call. Here's what I'm running:
```
brake crucible:execute[http://measure.eval.kanvix.com/cqf-ruler/baseDstu3,stu3,vte-submit-data]
```

And the output:
```
PASS SETUP: 
FAIL 01-Measure-Submit-Data: Bad response code: expected 200, but found 400 (URL: http://measure.eval.kanvix.com/cqf-ruler/baseDstu3/Measure/15821/$submit-data)
PASS TEARDOWN:
```

**Comment by mgramigna** [comment]
Upon doing some investigation into the 400, I realized it was possible to encounter a 409 during teardown because of the order I was deleting resources that depended on each other.

I also forgot some prefixes in the `relatedArtifact` section of some of the VTE libraries so I made those changes too.

As for the 400, it's pretty weird. For one, it only happens on Kanvix/the master branch of `cqf-ruler`. It does not happen on the `connectathon-21` branch. I even tried running the TestScript twice in a row on a brand new server instance with no resources on it at all. It passed the first time, successfully deleted all the resources, and then failed the second time. Running a `mvn clean` and then restarting the server makes the TestScript pass initially again.

This is the error message from the server: "Resource Practitioner/MitreTestScript-80582aa7-4f2a-4024-b1ef-ae83cc8d63a3 is deleted, specified in path: Encounter.participant.individual"

So the Encounter references the Practitioner, but why would we only see this on subsequent runs when it POSTs fine initially? /shrug

It's also strange that this doesn't happen on the `connectathon-21` branch. But I think they did add some fancy caching behavior or something. I'm not sure what's causing the error on Kanvix but maybe it has something to do with the existing caching behavior or something.

**Review by tstrass — CHANGES_REQUESTED** [review]
IIRC I think we decided yesterday we ought to assert that we can GET resources submitted after the call to `$submit-data` in order to assert that the server handled the submit correctly.

**Comment by mgramigna** [comment]
@tstrass I'd be fine adding this, although one thing I did come across since that discussion is I think those checks might be a little redundant. I believe submit data will fail if any of the resource were not successfully POSTed. This is the output from doing the operation on Postman:

image

I could still add it though to maybe give more insight to the user on which resource specifically in the parameters wasn't added correctly. Thoughts?

**Comment by tstrass** [comment]
Ah ok if that's the response to `$submit-data` then maybe we just assert that all of the expected resources are in the response body, and that their status is 200 OK?

**Comment by mgramigna** [comment]
Just pushed up some assertions to look for the resource IDs in the response body from submit-data. Also made all those resource IDs TestScript variables instead of hardcoded strings.

**Review by tstrass — APPROVED** [review]

**Review by okeefm — APPROVED** [review]
I like the changes. Tested a couple of times on my local cqf-ruler, and it works every time (no 400s).
